src/components/explore/AggregationsOverTimeGraph.svelte

```diff
@@ -143,6 +143,96 @@
   }
+
+  const getYTicks = (ranges) => {
+    if (
+      data[0].metric_type === 'histogram-linear' ||
+      data[0].metric_type === 'histogram-exponential'
+    ) {
+      // when the range is too small we need to custom set it
+      if (ranges[ranges.length - 1] < 5) return [0, 1, 2, 3, 4, 5];
+      return undefined;
+    }
+    if (
+      $store.activeBuckets.length &&
+      $store.proportionMetricType === 'proportions'
+    ) {
+      if (ranges[ranges.length - 1] - ranges[0] < 0.05)
+        return [0, ranges[ranges.length - 1]];
+    }
+    if (
+      $store.activeBuckets.length &&
+      $store.proportionMetricType === 'counts'
+    ) {
+      if (ranges[ranges.length - 1] < 5) return [0, 5];
+    }
+    // make sure that we have at least 5 elements in the tick array
+    if (ranges.length < 5 && !$store.activeBuckets.length) return yValues;
+    return undefined;
+  };
+
+  const getYDomain = (visiblePercentiles, buckets) => {
+    let yData = [];
+    let yDomainValues = [];
+
+    // linear and log graphs
+    if (
+      data[0].metric_type === 'histogram-linear' ||
+      yScaleType === 'scalePoint'
+    ) {
+      // eslint-disable-next-line no-restricted-syntax, guard-for-in
+      for (const p in visiblePercentiles) {
+        yData = [
+          ...yData,
+          ...data.map((arr) => arr.percentiles[visiblePercentiles[p]]),
+        ];
+      }
+    }
+    // categorical graphs
+    if (
+      buckets.length &&
+      data[0].metric_type !== 'histogram-linear' &&
+      yScaleType !== 'scalePoint'
+    ) {
+      if ($store.proportionMetricType === 'proportions') {
+        // eslint-disable-next-line no-restricted-syntax, guard-for-in
+        for (const bucket in buckets) {
+          yData = [
+            ...yData,
+            ...data.map((arr) => arr.proportions[buckets[bucket]]),
+          ];
+        }
+      } else if ($store.proportionMetricType === 'counts') {
+        // eslint-disable-next-line no-restricted-syntax, guard-for-in
+        for (const bucket in buckets) {
+          yData = [...yData, ...data.map((arr) => arr.counts[buckets[bucket]])];
+        }
+      }
+    }
+
+    yDomainValues = _.uniq(yData).sort((a, b) => a - b);
+    yDomainValues = yDomainValues.filter((a) => !Number.isNaN(a));
+
+    if (yScaleType === 'linear' && data[0].metric_type === 'histogram-linear')
+      return yDomainValues[yDomainValues.length - 1]
+        ? [yDomainValues[0], yDomainValues[yDomainValues.length - 1]]
+        : [NaN, NaN];
+
+    if (
+      data[0].metric_type !== 'histogram-linear' &&
+      yScaleType !== 'scalePoint' &&
+      buckets.length
+    ) {
+      if ($store.proportionMetricType === 'proportions') {
+        yDomainValues = yDomainValues.filter((a) => a < 1);
+        return [yDomainValues[0], yDomainValues[yDomainValues.length - 1]];
+      }
+      if ($store.proportionMetricType === 'counts')
+        return [yDomainValues[0], yDomainValues[yDomainValues.length - 1]];
+    }
```

Comment:
It seems like these are doing about the same thing and could be combined. Same with linear and histogram-linear?

Reply:
you're right. thanks for catching this